src/main/java/dev/j3fftw/litexpansion/items/WateringCan.java

```diff
@@ -0,0 +1,209 @@
+package dev.j3fftw.litexpansion.items;
+
+import dev.j3fftw.litexpansion.Items;
+import dev.j3fftw.litexpansion.LiteXpansion;
+import dev.j3fftw.litexpansion.utils.Constants;
+import dev.j3fftw.litexpansion.utils.Utils;
+import io.github.thebusybiscuit.slimefun4.core.handlers.ItemUseHandler;
+import io.github.thebusybiscuit.slimefun4.implementation.items.SimpleSlimefunItem;
+import me.mrCookieSlime.Slimefun.Lists.RecipeType;
+import me.mrCookieSlime.Slimefun.api.BlockStorage;
+import me.mrCookieSlime.Slimefun.cscorelib2.chat.ChatColors;
+import org.bukkit.Color;
+import org.bukkit.Effect;
+import org.bukkit.FluidCollisionMode;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.NamespacedKey;
+import org.bukkit.Particle;
+import org.bukkit.Sound;
+import org.bukkit.Tag;
+import org.bukkit.TreeType;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockFace;
+import org.bukkit.block.data.Ageable;
+import org.bukkit.block.data.BlockData;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemFlag;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.bukkit.inventory.meta.PotionMeta;
+import org.bukkit.persistence.PersistentDataType;
+import org.bukkit.util.RayTraceResult;
+
+import java.util.List;
+
+public class WateringCan extends SimpleSlimefunItem<ItemUseHandler> {
+
+    private static final int USE_INDEX = 8;
+    private static final int MAX_USES = 10;
+    private static final NamespacedKey usageKey = new NamespacedKey(LiteXpansion.getInstance(), "watering_can_usage");
+
+    public WateringCan() {
+        super(Items.LITEXPANSION, Items.WATERING_CAN, RecipeType.ENHANCED_CRAFTING_TABLE, new ItemStack[] {
+            Items.REFINED_IRON, null, Items.REFINED_IRON,
+            Items.REFINED_IRON, new ItemStack(Material.BUCKET), Items.REFINED_IRON,
+            null, Items.REFINED_IRON, null
+        });
+    }
+
+    @Override
+    public ItemUseHandler getItemHandler() {
+        return e -> {
+            e.cancel();
+
+            Player p = e.getPlayer();
+            RayTraceResult rayResult = p.rayTraceBlocks(5d, FluidCollisionMode.SOURCE_ONLY);
+
+            if (rayResult != null) {
+
+                Block b = rayResult.getHitBlock();
+                Location blockLocation = b.getLocation();
+                ItemStack item = e.getItem();
+                BlockData blockData = b.getBlockData();
+
+                // Fill if it hits water
+                if (b.getType() == Material.WATER) {
+                    updateUses(p, item, 2);
+
+                } else {
+
+                    if (b.getType() == Material.SUGAR_CANE) {
```

Comment:
This can be merged into an `else if` to reduce the indentation and nesting here.